sw/device/boot_rom/rom_crt.S

```diff
@@ -94,49 +94,17 @@ _start:
   li   t0, 0x55aa
   sw   t0, EDN_CTRL_REG_OFFSET(a0)
 
-  // Request memory scrambling and init
-  li a0, TOP_EARLGREY_SRAM_CTRL_MAIN_REGS_BASE_ADDR
-  li a1, (1<<SRAM_CTRL_CTRL_RENEW_SCR_KEY_BIT)|(1<<SRAM_CTRL_CTRL_INIT_BIT)
-  sw a1, SRAM_CTRL_CTRL_REG_OFFSET(a0)
-
   // Zero out the `.bss` segment.
   la   a0, _bss_start
   la   a1, _bss_end
   call crt_section_clear
 
-  // This zero out stack function is now handled by the hardware memory init from above
-  //
-  // As the stack grows downwards and we zero going forwards the start pointer
-  // starts as _stack_start and the end pointer at _stack_end.
-  //la   a0, _stack_start
-  //la   a1, _stack_end
-  //call crt_section_clear
-
   // Initialize the `.data` segment from the `.idata` segment.
   la   a0, _data_start
   la   a1, _data_end
   la   a2, _data_init_start
   call crt_section_copy
 
-  // Clobber all temporary registers.
```

Comment:
ah for DV purposes you MIGHT have to keep this.
It doesn't happen with all processors, but at least on some if you don't do this kind of clobbering, the "unknown values" can sometimes cause the simulation to get stuck.

Reply:
Curious - was that because on those processors, the registers were somehow not initialized to 0s on reset and they got stuck because of X-prop?

Reply:
yeah that's right. I think on those the registers were not reset at all, so the x's would just happily flow through address computations and eventually things would get stuck after a handful of operations. 

i think i've seen this on 3 separate arm processors.  I honestly don't remember if i ever saw it on ibex.

Reply:
Perhaps they used non-resettable flops for low power designs. On Ibex, the register file seems to be[ reset to 0](https://github.com/lowRISC/ibex/blob/1bbe27effeda63f34c4f5c06cc88da58f9c5a404/rtl/ibex_register_file_ff.sv#L58), so maybe this is ok?